python/sdss_access/path/path.py

```diff
@@ -716,3 +721,41 @@ def definitiondir(self, filetype, **kwargs):
 
 class AccessError(Exception):
     pass
+
+
+def _extract(template, example):
+
+    # escape the envvar $, any dots, and forward slashes
+    subtemp = template.replace('$', '\\$') \
+                      .replace('.', '\\.') \
+                      .replace('/', '\/')
+
+    # define named search pattern.
+    named_search = re.sub('{(\w+)}', '(?P<\\1>[\\w\\d+-_]+)', subtemp)
+
+    # fix duplicates by prepending '_' to them.
```

Comment:
Since repeated keywords in the template specify the same variable, we will never have two of the same keywords in a given template have different values.  It's probably safe to keep only one of the instances, and drop the `_xxx` duplicate.